aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -298,6 +295,18 @@ static void zero_numel_tensor_resize(Tensor& result, Tensor& result_indices,
       }
     }
   }
+  return sizes;
+}
+
+// Resize the result tensor and indices when result.numel() == 0 depending on values of
+// dim and keepdim for returning tensors containing reduction results.
+// This function should be called when you are reducing a zero-numel tensor and want to
+// resize the output and return it. This function exists for resizing zero-numel
+// tensors when the size of the reduction dimension is non-zero.
+static void zero_numel_tensor_resize(Tensor& result, Tensor& result_indices,
```

Comment:
you probably should take result as `const Tensor&` here

Reply:
This function already existed (apparently, `at::mode` is using it with non-const result). What I did was to extract `get_zero_numel_tensor_size` out of it. Should I change it nonetheless?

Reply:
You don't have to if you don't want to in this PR, but `const Tensor&` argument will work anywhere `Tensor&` used to work, and will save a const cast from a site where you only have a `const Tensor&`. It is not a big deal but just letting you know what the terminal state should look like.

Reply:
Ah, sorry. You are right. Since `at::mode` is the only function using it, I think I might as well get rid of this function.